### PR TITLE
Added publish with Delay And Expiry

### PR DESCRIPTION
**Pull request by Anjira-Singh**

### CODE REVIEW

File: src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java [hunk 1]
```diff
@@ -131,6 +130,31 @@ private int getShardId() {
...
+                    .headers(Collections.singletonMap("x-delay", delayMilliseconds))
+                    .deliveryMode(2)
+                    .build();
+        }
+    }
+
+    private AMQP.BasicProperties getPropertiesWithExpiry(AMQP.BasicProperties properties, final long expiryInMs){
+        if (expiryInMs > 0) {
+            val expiresAt = Instant.now().toEpochMilli() + expiryInMs;
+            properties = properties.builder()
```
[the remark this answers is not in the document]
Comment (Anjira-Singh) [reply src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java]: done

File: src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java [hunk 2]
```diff
@@ -131,6 +130,31 @@ private int getShardId() {
...
     }
 
+    private AMQP.BasicProperties getPropertiesWithDelay(final long delayMilliseconds){
+        if(config.getDelayType() == DelayType.TTL){
+            return new AMQP.BasicProperties.Builder()
+                    .expiration(String.valueOf(delayMilliseconds))
+                    .deliveryMode(2)
+                    .build();
+        }
+        else {
```
[the remark this answers is not in the document]
Comment (Anjira-Singh) [reply src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java]: removed this

File: src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java [hunk 3]
```diff
@@ -75,24 +75,23 @@ public final void publishWithDelay(final Message message, final long delayMillis
...
-            properties = properties.builder()
-                    .headers(ImmutableMap.of(MESSAGE_EXPIRY_TEXT, expiresAt))
-                    .build();
-        }
-        publish(message, properties);
+        val finalProperties = getPropertiesWithExpiry(properties, expiryInMs);
+        publish(message, finalProperties);
+    }
+
+    public final void publishWithDelayAndExpiry(final Message message, final long expiryInMs, final long delayMilliseconds)
```
[the remark this answers is not in the document]
Comment (Anjira-Singh) [reply src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java]: done

File: src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java [hunk 4]
```diff
@@ -131,6 +130,31 @@ private int getShardId() {
...
+        }
+        else {
+            return new AMQP.BasicProperties.Builder()
+                    .headers(Collections.singletonMap("x-delay", delayMilliseconds))
+                    .deliveryMode(2)
+                    .build();
+        }
+    }
+
+    private AMQP.BasicProperties getPropertiesWithExpiry(AMQP.BasicProperties properties, final long expiryInMs){
```
[the remark this answers is not in the document]
Comment (Anjira-Singh) [reply src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java]: done

File: src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java [hunk 5]
```diff
@@ -131,6 +130,31 @@ private int getShardId() {
...
+        if (expiryInMs > 0) {
```
[the remark this answers is not in the document]
Comment (Anjira-Singh) [reply src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java]: done

**Review by r0goyal — APPROVED** [review]

**Pull request merged**